src/util/events.rs

```diff
@@ -92,7 +92,10 @@ pub enum Event {
 	/// Used to indicate that ChannelManager::process_pending_htlc_forwards should be called at a
 	/// time in the future.
 	PendingHTLCsForwardable {
-		/// The amount of time that should be waited prior to calling process_pending_htlc_forwards
+		/// The minimum amount of time that should be waited prior to calling
+		/// process_pending_htlc_forwards. To increase the effort required to correlate payments,
```

Comment:
But if you're already on the payment path, you can already do decorrelation attacks with hashes ? You're trying to break what kind of analysis here ?

Reply:
Hmm...I guess I haven't formalized it, but there's almost certainly some stuff you can learn if the timing is super, super consistent. eg you could learn the number of hops that something took just by looking at the timing. Hiding that at least somewhat or for individual payments.

Reply:
I would argue if you're two spots on the same payment path, you can guess the number of hops by recomputing the per-hop fees between your A and your B given fees are public. Out-of-topic, as you said if user is willingly to wait shouldn't hurt.

Reply:
Right, but a user could have a more privacy-conscious routing algorithm, and we don't want to reveal info unless we have to. I think its generally a good idea, but agreed more study and a formal threat model would be better.